### PR TITLE
Use modal for course registration exam confirmation

### PR DESCRIPTION
**Pull request by ungaralex**
### Description

As discussed with @krusche , the current confirmation dialog for PGdP students when registering for the course is relatively ugly. Since we want to have standardized components anyway, it would be nice to have a confirmation dialog in form of a modal here.
I added a new shared jhi-confirm-button component, since we only had our generic button component before that (the confirm button uses our generic button combined with a modal). The registration now uses this confirmation button for the PGdP course.

### Screenshots

Screenshot_20191015_001048

### CODE REVIEW

File: src/main/webapp/i18n/de/student-dashboard.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
                 "noCourse": "Keine Kurse für eine Registrierung verfügbar",
                 "courseLoading": "Kurse werden geladen...",
                 "pleaseSelectCourse": "# Wähle einen Kurs aus",
-                "examSignupConfirmation": "Mit Anmeldung zu diesem Kurs erfolgt eine verbindliche Anmeldung zur Prüfung."
+                "examSignupConfirmation": "Mit Anmeldung zu diesem Kurs erfolgt eine verbindliche Anmeldung zur Prüfung.",
+                "examSignupConfirmationTitle": "Bestätigung der Klausurteilnahme"
```
Comment (krusche) [inline src/main/webapp/i18n/de/student-dashboard.json]:
```suggestion
                "examSignupConfirmationTitle": "Bestätigung der Prüfungsteilnahme"
```

File: src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            <h4 class="modal-title">{{ title | translate }}</h4>
+            <button type="button" class="close" aria-label="Close button" aria-describedby="modal-title" (click)="modal.dismiss('Cross click')">
+                <span aria-hidden="true">&times;</span>
+            </button>
+        </div>
+        <div class="modal-body">
+            <p>{{ text | translate }}</p>
+        </div>
+        <div class="modal-footer">
+            <button type="button" class="btn btn-outline-secondary" (click)="modal.dismiss('cancel click')">Cancel</button>
```
Comment (krusche) [inline src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts]: do we need translations here?

File: src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            <button type="button" ngbAutofocus class="btn btn-danger" (click)="modal.close('Ok click')">Ok</button>
```
Comment (krusche) [inline src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts]: I would call this Confirm.
Do we need translations here?

File: src/main/webapp/app/components/confirm-button/confirm-button.component.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import { Component, EventEmitter, OnInit, Output } from '@angular/core';
 
+// TODO this can probably be moved to one of our generic delete components
 @Component({
-    selector: 'jhi-confirm-button',
+    selector: 'jhi-confirm-delete-button',
```
Comment (krusche) [inline src/main/webapp/app/components/confirm-button/confirm-button.component.ts]: why did you change this?

File: src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts (hunk 2 above)
Comment (ungaralex) [reply src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts]: I would argue that "Cancel" and "Ok" are both valid in German and English. Although, if there will be other languages in the future, those values won't show up. I'll add translations so that we are safe

File: src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts (hunk 3 above)
Comment (ungaralex) [reply src/main/webapp/app/shared/components/confirm-autofocus-button.component.ts]: I changed the translation to "Confirm"

File: src/main/webapp/app/components/confirm-button/confirm-button.component.ts (hunk 4 above)
Comment (ungaralex) [reply src/main/webapp/app/components/confirm-button/confirm-button.component.ts]: Functionality-wise this was a delete button. It had "Delete" as the string and a confirmation, before there was confirmation event fired. We should use the new generic delete component in this case anyway, so I now completely removed this component (we only used it for deleting lecture attachments) and switched the usage in related components to our new `jhiDeleteButton` directive
